app/controllers/v1/inventories_controller.rb

```diff
@@ -0,0 +1,39 @@
+class V1::InventoriesController < ApplicationController
+
+  before_action :authenticate_user!
+
+  def index
+    @inventories = Inventory.where(user: current_user)
+  end
+
+  def create
+    @inventory = Inventory.new
+    @inventory.name = inventory_params[:name]
+    unless inventory_params[:deadline].blank?
+      begin
+        @inventory.deadline = DateTime.strptime(inventory_params[:deadline], '%m/%d/%Y')
+      rescue
+        @inventory.errors.add(:deadline, 'must be in DD/MM/YYYY format')
+        return render_error
+      end
+    end
+    @inventory.district = District.find(inventory_params[:district][:id])
+    @inventory.user = current_user
```

Comment:
you can also do `@inventory = current_user.inventories.new(inventory_params.extract!(:name))`